openpype/plugins/publish/collect_transient_data_dir.py

```diff
@@ -0,0 +1,105 @@
+"""
+Requires:
+    anatomy
+
+
+Provides:
+    instance.data     -> stagingDir (folder path)
+                      -> stagingDirPersistence (bool)
+                      -> stagingDir_persistent (bool) [deprecated]
+"""
+import copy
+import os.path
+
+import pyblish.api
+
+from openpype.pipeline import get_transient_data_profile
+from openpype.lib import StringTemplate
+
+
+class CollectTransientDataDir(pyblish.api.InstancePlugin):
+    """Apply matching Transient data profile (custom stagingDir) to a instance.
+
+    Transient dir could be useful in specific use cases where is
+    desirable to have temporary renders in specific, persistent folders, could
+    be on disks optimized for speed for example.
+
+    It is studio responsibility to clean up obsolete folders with data.
+
+    Location of the folder is configured in `project_anatomy/templates/others`.
+    ('transient' key is expected, with 'folder' key)
+
+    Which family/task type/subset is applicable is configured in:
+    `project_settings/global/transient_data_profiles/profiles`
+
+    Deprecated path (backward compatibility):
+    `project_settings/global/tools/publish/custom_staging_dir_profiles`
+    """
+    label = "Collect Transient Data Directory"
+    order = pyblish.api.CollectorOrder + 0.4990
+
+    template_key = "transient"
+
+    def process(self, instance):
+        family = instance.data["family"]
+        subset_name = instance.data["subset"]
+        host_name = instance.context.data["hostName"]
+        project_name = instance.context.data["projectName"]
+        project_settings = instance.context.data["project_settings"]
+        anatomy = instance.context.data["anatomy"]
+        task = instance.data["anatomyData"].get("task", {})
+
+        transient_data_profile = get_transient_data_profile(
+            project_name, host_name, family, task.get("name"),
+            task.get("type"), subset_name, project_settings=project_settings,
+            anatomy=anatomy, log=self.log)
+
+        if transient_data_profile:
```

Comment:
Shouldn't this be `if not` ? Otherwise the code at line 67 will never execute?  Or we'll pass an empty `transient_data_profile` to `_apply_transient_data` (and we are not doing any check there if it's empty or not)

Reply:
huh.. yes that is correct